examples/auth/index.js

```diff
@@ -127,7 +127,7 @@ app.post('/login', function (req, res, next) {
 });
 
 /* istanbul ignore next */
-if (!module.parent) {
+if (module.children) {
```

Comment:
I'm not sure this change is correct looking at the docs (even though it happens to work for our tests). The current conditional tests to see if this file was required by another module or not (i.e. that it is the application entry point -- the desired test), but this change instead checks if any module require by this module was the first time it was required or not. That seems like it would be a side effect, and isn't the test this line is trying to test for. It would seem that you need to construct a test using `require.main` to correctly replicate this line as intended.

Reply:
Thank you for taking the time for such a detailed reply.I'm sorry for my mistake.Follow your instructions, I have tested module.children and require.main, and I think you are right. Personally, I think it is more appropriate to use require.main instead of module.parent in this example